core/chain-configs/src/genesis_config.rs

```diff
@@ -164,7 +174,16 @@ impl GenesisConfig {
     /// GenesisConfig structure.
     pub fn from_file<P: AsRef<Path>>(path: P) -> Self {
         let reader = BufReader::new(File::open(path).expect("Could not open genesis config file."));
-        serde_json::from_reader(reader).expect("Failed to deserialize the genesis records.")
+        #[allow(unused_mut)]
+        let mut genesis_config: GenesisConfig =
+            serde_json::from_reader(reader).expect("Failed to deserialize the genesis records.");
+        #[cfg(feature = "protocol_feature_evm")]
+        {
+            genesis_config.runtime_config.transaction_costs.evm_config = EvmCostConfig::default();
```

Comment:
is this actually needed? If `evm` is enabled shouldn't the deserializer fill in the default value automatically?

Reply:
Right now we just skip it with `#serde(skip)`, but because it's part of the feature we probably can deserialize it.
Updated